modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java

```diff
@@ -759,6 +761,150 @@ public void shouldGenerateDefaultValueForEnumRequestParameter() throws IOExcepti
 
     }
 
+
+
+    @Test
+    public void shouldGenerateOptionalForPojos_nullabilityEnabled_optionalEnabled() throws IOException {
+        File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
+        output.deleteOnExit();
+        String outputPath = output.getAbsolutePath().replace('\\', '/');
+
+        final OpenAPI openAPI = TestUtils.parseFlattenSpec("src/test/resources/3_0/spring/issue_11383.yaml");
+        final SpringCodegen codegen = new SpringCodegen();
+        codegen.setOpenAPI(openAPI);
+        codegen.setOutputDir(output.getAbsolutePath());
+
+        codegen.additionalProperties().put(SpringCodegen.DELEGATE_PATTERN, "true");
+        codegen.additionalProperties().put(AbstractJavaCodegen.OPENAPI_NULLABLE, "true");
+        codegen.additionalProperties().put(OptionalFeatures.USE_OPTIONAL, "true");
+
+        ClientOptInput input = new ClientOptInput();
+        input.openAPI(openAPI);
+        input.config(codegen);
+
+        DefaultGenerator generator = new DefaultGenerator();
+
+        generator.setGeneratorPropertyDefault(CodegenConstants.MODELS, "true");
+        generator.setGeneratorPropertyDefault(CodegenConstants.MODEL_TESTS, "false");
+        generator.setGeneratorPropertyDefault(CodegenConstants.MODEL_DOCS, "false");
+        generator.setGeneratorPropertyDefault(CodegenConstants.APIS, "true");
+        generator.setGeneratorPropertyDefault(CodegenConstants.SUPPORTING_FILES, "false");
+
+        generator.opts(input).generate();
+
+        assertFileContains(Paths.get(outputPath + "/src/main/java/org/openapitools/model/Params.java"),
+                "private Optional<String> optionalString = Optional.empty();",
+                "private JsonNullable<String> optionalNullableString = JsonNullable.undefined();",
+                "private String requiredNullableString;");
```

Comment:
So this would mean that this is "fixing" a current bug - or at least a current behaviour.

I believe the generator currently creates required nullable fields with `JsonNullable` - so this would be a change in behaviour.